watchtower-plugin/src/net/http.rs

```diff
@@ -502,6 +522,75 @@ mod tests {
         assert!(matches!(error, AddAppointmentError::ApiError { .. }));
     }
 
+    #[tokio::test]
+    async fn test_request() {
+        let mut server = mockito::Server::new_async().await;
+        let api_mock = server
+            .mock("POST", Endpoint::Register.path().as_str())
+            .with_status(200)
+            .with_header("content-type", "application/json")
+            .create_async()
+            .await;
+
+        let response = request(
+            &NetAddr::new(server.url()),
+            Endpoint::Register,
+            &None,
+            Method::POST,
+            Some(json!({ "":"" })),
+        )
+        .await
+        .unwrap();
+
+        api_mock.assert_async().await;
+        assert!(matches!(response, Response { .. }));
+    }
+    #[tokio::test]
+    async fn test_request_connection_error() {
+        assert!(matches!(
+            request(
+                &NetAddr::new("http://unreachable_url".to_owned()),
+                Endpoint::Register,
+                &None,
+                Method::POST,
+                Some(json!({ "":"" })),
```

Comment:
This is equivalent to `json!("")`

Reply:
but I'm getting error! Mismatch error. 
```
mismatched types
expected enum `std::option::Option<_>`
   found enum `serde_json::Value

Reply:
`json!({ "":"" })` == `json!("")`

So you need to do `Some(json!(""))`

Reply:
Ohh 😅😅